rdtools/soiling.py

```diff
@@ -701,3 +704,176 @@ def soiling_srr(energy_normalized_daily, insolation_daily, reps=1000,
+        Typically ``soiling_interval_summary['stochastic_soiling_profiles']`` obtained with
+        :py:func:`rdtools.soiling.soiling_srr` or :py:meth:`rdtools.soiling.SRRAnalysis.run`
+
+    confidence_level : float, default 68.2
+        The size of the confidence interval to use in determining the upper and lower
+        quantiles reported in the returned DataFrame. (The median is always included in
+        the result.)
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame describing annual soiling rates.
+        Columns:
+
+        * `year`
+        * `soiling_ratio_median` - The median insolation-weighted soiling ratio for the year
+        * `soiling_ratio_low` - The lower edge of the confidence interval for
+          insolation-weighted soiling ratio for the year
+        * `soiling_ratio_high` - The upper edge of the confidence interval for
+          insolation-weighted soiling ratio for the year
+    '''
+
+    df = pd.DataFrame(pd.concat(stochastic_soiling_profiles))
+
+    ci_quantiles = [0.5 - confidence_level/2/100, 0.5 + confidence_level/2/100]
+    annual_soiling = df.groupby(df.index.year).quantile([0.5, ci_quantiles[0], ci_quantiles[1]])
+    annual_soiling.index.rename(['year', 'quantile'], inplace=True)
+    annual_soiling.columns = ['insolation_weighted_soiling_ratio']
+
+    # Make the output look like monthly_soiling_rates
+    annual_soiling = annual_soiling.reset_index(['quantile'])
+    annual_soiling = pd.pivot(annual_soiling, columns='quantile')
+    annual_soiling.columns = annual_soiling.columns.droplevel(0)
+    annual_soiling.columns.name = None
+    annual_soiling = annual_soiling.sort_index(axis=1)
+    annual_soiling.columns = ['soiling_ratio_low', 'soiling_ratio_median', 'soiling_ratio_high']
+    annual_soiling = annual_soiling[['soiling_ratio_median', 'soiling_ratio_low', 'soiling_ratio_high']]
+    annual_soiling = annual_soiling.reset_index()
+
+    return annual_soiling
+
+
+def monthly_soiling_rates(soiling_interval_summary, min_length=14,
+                          max_relative_slope_error=500.0, reps=100000,
+                          confidence_level=68.2):
+    '''
+    Use Monte Carlo to calculate typical monthly soiling rates. Samples possible
+    soiling rates from soiling rate confidence intervals associated with soiling
+    intervals assuming a uniform distribution. Soiling intervals get samples
+    proportionally to their overlap with each calendar month.
+
+    Parameters
+    ----------
+    soiling_interval_summary : pd.DataFrame
+        DataFrame describing soiling intervals. Typically from
+        ``soiling_interval_summary['confidence_interval']`` obtained with
+        :py:func:`rdtools.soiling.soiling_srr` or
+        :py:meth:`rdtools.soiling.SRRAnalysis.run` Must have columns
+        ``slope_high``, ``slope_low``, ``slope``, ``length``, ``valid``,
+        ``start``, and ``end``.
+
+    min_length : int, default 14
```

Comment:
It is very similar. In this case we want to keep the minimum length for intervals we consider longer than 2, because when looking at monthly soiling rates, we want to be more restrictive than when we we are doing a long term soiling ratio calculation. 

It would probably be a good idea to rename this parameter `min_interval_length`, but would that be confusing if we leave it to have a different default than the "other" `min_interval_length`?

Reply:
Fine with me to have the same name, should the docstring mention the reason for the difference in default value?

Reply:
Yep, good suggestion